app/models/package_manager/maven/maven_central.rb

```diff
@@ -20,4 +20,20 @@ def self.one_version(raw_project, version_string)
   def self.missing_version_remover
     PackageManager::Base::MissingVersionRemover
   end
+
+  # maven-metadata.xml for Maven Central does not appear to be guaranteed to contain all relevant versions for a package
```

Comment:
What assumptions are being made here & can we validate/invalidate those assumptions (e.g. log when those assumptions are no longer true)?

Reply:
What are you referring to in this situation, exactly? Are you referring to how we know that the metadata isn't guaranteed to contain all versions? If so, we have determined that to be true from some instances of this that we've noticed already, where a metadata file did not contain all the versions listed in the index.

If you are referring to the assumptions being made in terms of the HTML format we're parsing, then yes, perhaps there are some things we could do to log if we don't get what we expect.

Reply:
Stepping back to look at the overall Maven version data landscape - This method is adding another source of where we get version data. This means we can now get version data from at least three places:
1.) Version data from the maven directory
2.) Version data from the maven .xml file
3.) Version data stored in libraries already

We have discovered that 2 has data issues, and we are currently trying to rectify data issues around 3. We have looked at 1 and have come to the conclusion that the data seems more accurate than than 2 - but to my knowledge, we've only checked a small % of all Maven packages, which may or may not be a representative sample. Is there anything we can check or cross reference to reinforce or disprove the conclusion that a) 1 is always better than 2 and b) 1 is the most correct source of maven version data? Could we do something to let us know if things seem to be trending in the wrong direction / let us know that this data source is decaying and we should look for another source?

Reply:
Yup; I think that we can trust that this data source is likely correct for us to use in the short term, but agreed it would be good to add some logging about assumptions we're making so that we can reassess in the future if needed. I can do that.

Reply:
An additional note to make: This is not exactly replacing source (2) above with source (1). Functionally, we already depend on the directory structure from source (1) in order to get the data from the POMs. So really this new directory structure parsing isn't new, as we already depended on these directories existing in the previous implementation. All this PR is doing is removing source (2) in favor of source (1) which we already use.

Reply:
We discussed this further and decided that these logs for the diff should be unnecessary. Since we already use the directory structure to find the POMs we need, and we are using the same directory structure here, we are necessarily going to find equal or more versions using this approach than with using the metadata.